File: Data Structures & Algorithms/validate-parentheses/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
       if (s.length === 1) return false;

        const bracket = {']' : '[', '}':'{', ')':'('};
        const stack = [];
        let r = 0, n = s.length;

        while (r < n){
            if (bracket[s[r]]){
                if (stack.pop() !== bracket[s[r]]) return false;
            }else {
                stack.push(s[r]);
            }
            r++;
        }

        return stack.length === 0;
    }
}

File: Data Structures & Algorithms/baseball-game/submission-0.java
class Solution {
    public int calPoints(String[] operations) {
        Stack<Integer> records = new Stack<>();

        for (String op : operations){
            if (op.equals("+")){
                int top = records.pop();
                int sum = top + records.peek();

                records.push(top);
                records.push(sum); 
            }else if (op.equals("D")){
                records.push(2 * records.peek());
            }else if (op.equals("C")){
                records.pop();
            }else {
                records.push(Integer.parseInt(op));
            }
        }

        int sum = 0;

        for (int n : records){
            sum += n;
        }

        return sum;

    }
}